### PR TITLE
Update and clean up vSphere Cloud Provider docs

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>
Co-authored-by: Cormac Hogan <chogan@vmware.com>

**What this PR does / why we need it**:
Made some improvements to the vSphere cloud provider docs with @cormachogan. Mainly organizes the docs so it's easier to consume and dives into more details on how to install / setup the out-of-tree cloud provider and the vSphere CSI driver. More updates to come in follow-up PRs.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/book/cloud_provider_interface.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+4) Install a Guest OS on all VMs. VMware recommends Ubuntu 18.04.1 LTS (Bionic Beaver) - 64-bit PC (AMD64) Server.
+5)Disable swap on all K8s worker nodes. Use the ‘swapoff -a’ command and then modify the /etc/fstab file and remove any swap entries.
+6) Ensure Docker is installed and running on all nodes. VMware recommends Docker version 18.06 CE.
+7) Install kubeadm version 1.14.2-00 on all master and worker nodes.
+
+Steps 8-10 are specifically for the master(s).
+8) Create an /etc/kubernetes/kubeadminit.yaml cluster configuration file on the master, and run kubeadm init to set up the master node. The manifest states the cloud-provider is external, though we have not yet set it up. If the command is successful, the output will the kubeadm join command to join the worker nodes to the cluster.
```
Comment (andrewsykim) [inline docs/book/cloud_provider_interface.md]: markdown verify doesn't like that this is 8) and think it should be 1) because of the line break.

**Comment by dvonthenen** [comment]
Content looks great! 👍Just have the markdown issue(s) caught by the CI.

File: docs/book/cloud_provider_interface.md (hunk 1 above)
Comment (andrewsykim) [reply docs/book/cloud_provider_interface.md]: @cormachogan I might convert these to regular bullet points (not numbered) cause it's failing mark down checks.

Comment (cormachogan) [reply docs/book/cloud_provider_interface.md]: Go for it @andrewsykim

Comment (andrewsykim) [reply docs/book/cloud_provider_interface.md]: @cormachogan, on second thought I think we should revisit this list, there are some thing there that are too prescriptive IMO. For now I'm going to have it point to https://cloud-provider-vsphere.sigs.k8s.io/tutorials/deploying_cloud_provider_vsphere_with_rbac.html and revisit this section later.

**Pull request merged**